src/components/views/story-editor-view/story-editor-view.tsx

```diff
@@ -90,6 +92,10 @@ const StoryEditorView: React.FC<Props> = ({
     }
   }, [ addingKnots, newPoint, onAddNode ]);
 
+  React.useEffect(() => {
+    editingEntityInfo ? engineRef.current.getModel().setLocked(true) : engineRef.current.getModel().setLocked(false);
+  }, [editingEntityInfo]);
```

Comment:
Remember, don't think too complicated, dynamically change only the part that needs to be changed.
```suggestion
  React.useEffect(() => {
    engineRef.current.getModel().setLocked(editingEntityInfo);
  }, [ editingEntityInfo ]);
```